### PR TITLE
Bug 1492018: Prefer `.blockIndicator` for styling block indicators

### PR DESCRIPTION
**Pull request by ExE-Boss**
See [bug 1492018](https://bugzil.la/1492018) for details.

Depends on https://github.com/mozilla/kuma/pull/4979 and ~~https://github.com/mozilla/kuma/pull/4962~~.

review?(@Elchi3, @jwhitlock, @wbamberg, @escattone)

### CODE REVIEW

**Comment by wbamberg** [comment]
As noted in https://bugzilla.mozilla.org/show_bug.cgi?id=1492018, @schalkneethling is probably the best person to review this.

File: macros/SeeCompatTable.ejs [hunk 1]
```diff
@@ -12,7 +12,6 @@ var str = mdn.localString({
     "ru"    : "<strong>Это экспериментальная технология</strong><br />Так как спецификация этой технологии ещё не стабилизировалась, смотрите <a href='#Browser_compatibility'>таблицу совместимости</a> по поводу использования в различных браузерах. Также заметьте, что синтаксис и поведение экспериментальной технологии может измениться в будущих версиях браузеров, вслед за изменениями спецификации."
 });
 
-
-%><div class="notice overheadIndicator experimental">
+%><div class="blockIndicator experimental">
```
Comment (ExE-Boss) [inline macros/SeeCompatTable.ejs]: This change depends on https://github.com/mozilla/kuma/pull/4962.

File: macros/minversionGeneric.ejs [hunk 2]
```diff
@@ -76,18 +76,18 @@ switch(lang) {
...
         break;
     case 'header':
-        %><div class="overheadIndicator standardNote standardNoteBlock">
+        %><div class="blockIndicator standardNote">
 <p><%-text%></p>
 </div><%
         break;
     case 'note':
-        %><div class="blockIndicator standardNote standardNoteBlock">
+        %><div class="blockIndicator standardNote">
```
Comment (schalkneethling) [inline macros/minversionGeneric.ejs]: Did not find any pages that uses this macro:
https://developer.mozilla.org/en-US/search?topic=none&kumascript_macros=minversionGeneric&locale=*

File: macros/non-standardGeneric.ejs [hunk 3]
```diff
@@ -50,7 +50,7 @@ switch($0) {
     %><span class="inlineIndicator nonStandard nonStandardInline"><%-str_inline%></span><%
     break;
   case 'header':
-    %><div class="overheadIndicator nonStandard nonStandardHeader">
+    %><div class="blockIndicator nonStandard nonStandardHeader">
```
Comment (schalkneethling) [inline macros/non-standardGeneric.ejs]: No pages found using this macro:
https://developer.mozilla.org/en-US/search?topic=none&kumascript_macros=non-standardGeneric&locale=*

File: macros/unimplementedGeneric.ejs [hunk 4]
```diff
@@ -63,7 +63,7 @@ switch($0) {
     %><span class="inlineIndicator unimplemented unimplementedInline"><%-str%></span><%
     break;
   case 'header':
-    %><div class="overheadIndicator unimplemented unimplementedHeader">
+    %><div class="blockIndicator unimplemented unimplementedHeader">
```
Comment (schalkneethling) [inline macros/unimplementedGeneric.ejs]: Did not find any pages using this macro:
https://developer.mozilla.org/en-US/search?topic=none&kumascript_macros=secureContextGeneric&locale=*

File: tests/macros/test-draft.js [hunk 5]
```diff
@@ -14,89 +14,89 @@ describeMacro('draft', function () {
     itMacro('No arguments (en-US)', function (macro) {
         return assert.eventually.equal(
             macro.call(),
-            `<div class="overheadIndicator draft">\n    <p><strong>Draft</strong><br/>\n    This page is not complete.</p>\n    \n</div>`
+            `<div class="blockIndicator draft">\n    <p><strong>Draft</strong><br/>\n    This page is not complete.</p>\n    \n</div>`
```
Comment (schalkneethling) [inline tests/macros/test-draft.js]: No pages found using this macro (the name of the file also suggest it was temporary):
https://developer.mozilla.org/en-US/search?topic=none&kumascript_macros=test-draft&locale=*

**Comment by schalkneethling** [comment]
Used this following pages to test:

* http://localhost:8000/en-US/docs/Mozilla/Projects/SpiderMonkey/JSAPI_reference/JS_CompileFileHandleForPrincipals
* http://localhost:8000/bn-BD/docs/Web/JavaScript/Reference/Global_Objects/Number
* http://localhost:8000/it/docs/Building_an_Extension
* http://localhost:8000/tr/docs/Web/JavaScript/Reference/Global_Objects/Object/observe
* http://localhost:8000/es/docs/Web/Guide/DOM
* http://localhost:8000/en-US/docs/Web/API/DeviceLightEvent/Using_light_sensors
* http://localhost:8000/ro/docs/Web/Apps/Quickstart

There is also a conflict with `macros/HTMLVersionHeader.ejs`

File: macros/SimpleBadge.ejs [hunk 6]
```diff
@@ -5,4 +5,4 @@
         $1 - Class name for custom style
         $2 - Tooltip text (optional)
 
-*/%><span class="badge <%= $1 %>" title="<%= $2 %>"><span class="badgeText"><%- $0 %></span></span>
+*/%><span class="inlineIndicator <%= $1 %>" title="<%= $2 %>"><span class="badgeText"><%- $0 %></span></span>
```
[the remark this answers is not in the document]
Comment (ExE-Boss) [reply macros/SimpleBadge.ejs]: SimpleBadge is used in other macros to generate inline indicators.

[the remark this answers is not in the document]
Comment (schalkneethling) [reply macros/SimpleBadge.ejs]: Ah of course. There may be a couple of the others that fit this same scenario.

File: macros/non-standardGeneric.ejs (hunk 3 above)
Comment (ExE-Boss) [reply macros/non-standardGeneric.ejs]: This is used inside `{{Non-standard_Inline}}` and `{{Non-standard_Header}}`

File: tests/macros/test-draft.js (hunk 5 above)
Comment (ExE-Boss) [reply tests/macros/test-draft.js]: 🤦🏻‍♂️ This is a `mocha` unit test file for ensuring that the `{{Draft}}` macro creates correct output.

File: macros/HTMLVersionHeader.ejs [hunk 7]
```diff
@@ -18,6 +18,6 @@ var s_msg = mdn.localString({
 
 var str = s_msg[0] + link + s_msg[1];
 %>
-<div class="overheadIndicator htmlVer htmlMinVerHeader">
+<div class="blockIndicator htmlVer htmlMinVerHeader">
```
[the remark this answers is not in the document]
Comment (ExE-Boss) [reply macros/HTMLVersionHeader.ejs]: This just got removed in https://github.com/mdn/kumascript/pull/927.

File: macros/SimpleBanner.ejs [hunk 8]
```diff
@@ -6,7 +6,7 @@
         $2 - Additional text to display
 
 */%>
-<div class="banner <%= $1 %>">
+<div class="blockIndicator <%= $1 %>">
```
[the remark this answers is not in the document]
Comment (ExE-Boss) [reply macros/SimpleBanner.ejs]: Same as with `{{SimpleBadge}}`.

File: macros/minversionGeneric.ejs (hunk 2 above)
Comment (ExE-Boss) [reply macros/minversionGeneric.ejs]: This is used in many minversion type macros.

File: macros/unimplementedGeneric.ejs (hunk 4 above)
Comment (ExE-Boss) [reply macros/unimplementedGeneric.ejs]: This is used inside the [`{{Unimplemented_Inline}}`](https://github.com/mdn/kumascript/blob/master/macros/unimplemented_inline.ejs) macro.

File: tests/macros/test-draft.js (hunk 5 above)
Comment (schalkneethling) [reply tests/macros/test-draft.js]: Doh! :) Right you are.

File: macros/HTMLVersionHeader.ejs (hunk 7 above)
[the remark this answers is not in the document]
Comment (schalkneethling) [reply macros/HTMLVersionHeader.ejs]: Ok, so a rebase will then remove it from this PR as well. Thanks!

**Comment by schalkneethling** [comment]
@ExE-Boss seeing that there is a number of macros here that are used by other macros. Can you suggest some pages I can use to test those other than the list of pages I specified in https://github.com/mdn/kumascript/pull/830#issuecomment-427803336

**Comment by ExE-Boss** [comment]
(In reply to @schalkneethling from https://github.com/mdn/kumascript/pull/830#issuecomment-427809741)
> @ExE-Boss seeing that there is a number of macros here that are used by other macros. Can you suggest some pages I can use to test those other than the list of pages I specified in [#830 (comment)](https://github.com/mdn/kumascript/pull/830#issuecomment-427803336)

I just created https://developer.mozilla.org/docs/Sandbox/Indicators specifically for this reason.

**Comment by ExE-Boss** [comment]
@schalkneeting: This is ready for a second round of review.

File: macros/SeeCompatTable.ejs [hunk 9]
```diff
@@ -12,7 +12,6 @@ var str = mdn.localString({
     "ru"    : "<strong>Это экспериментальная технология</strong><br />Так как спецификация этой технологии ещё не стабилизировалась, смотрите <a href='#Browser_compatibility'>таблицу совместимости</a> по поводу использования в различных браузерах. Также заметьте, что синтаксис и поведение экспериментальной технологии может измениться в будущих версиях браузеров, вслед за изменениями спецификации."
 });
 
-
-%><div class="notice overheadIndicator experimental">
+%><div class="blockIndicator experimental indicator-warning">
```
Comment (ExE-Boss) [inline macros/SeeCompatTable.ejs]: https://github.com/mozilla/kuma/pull/4979 now adds the following indicator classes:
- `.indicator-info` - grey
- `.indicator-version` - blue
- `.indicator-warning` - yellow
- `.indicator-danger` - red

**Comment by schalkneethling** [comment]
Thanks for the update @ExE-Boss, will review as soon as possible.

File: macros/es7.ejs [hunk 10]
```diff
@@ -18,6 +18,6 @@ var s = mdn.localString({
 });
 %>
 
-<div class="overheadIndicator" style="background:#9CF49C;">
+<div class="blockIndicator" style="background:#9CF49C;">
```
[the remark this answers is not in the document]
Comment (schalkneethling) [reply macros/es7.ejs]: This one is still displaying incorrectly. I noticed that you removed the selector that was looking for the `style` attribute with the specific colour value, but this macro should be updated as follows:

```
<div class="blockIndicator es-harmony">
```

**Comment by ExE-Boss** [comment]
@wbamberg: Can this be merged now?

**Comment by schalkneethling** [comment]
This still needs approval from someone with write access. @Elchi3 can you do that based on @jwhitlock comment here https://github.com/mozilla/kuma/pull/4979#issuecomment-430262972

**Comment by ExE-Boss** [comment]
This still needs merging from someone with write access.

**Pull request merged**